exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/DrillCompoundIdentifier.java

```diff
@@ -69,31 +70,38 @@ public void addIndex(int index, SqlParserPos pos){
     }
   }
 
-  public SqlNode getAsSqlNode(){
-    if(ids.size() == 1){
+  public SqlNode getAsSqlNode(Set<DrillCompoundIdentifier> fullSchemasSet) {
```

Comment:
It would be great don't convert original calcite `SqlNode` with `CompoundIdentifierConverter`. 
In that case unit tests from my PR would have passed successfully but drill functionality with nested complex schema wouldn't work (ex: quering json arrays). So I think we can't refuse from `DrillParserWithCompoundIdConverter` logic. 

And the main idea of this PR to improve that CompoundIdentifierConverter don't ignore full schema in the beginning of the identifier.
If I missed something please correct me.